draft-ietf-httpbis-client-hints.md

```diff
@@ -169,6 +200,17 @@ Implementers ought to consider both user and server controlled mechanisms and po
 Implementers SHOULD support Client Hints opt-in mechanisms and MUST clear persisted opt-in preferences when any one of site data, browsing history, browsing cache, or similar, are cleared.
 
 
+## Deployment and Security Risks
+Deployment of new request headers requires several considerations:
+
+  - Potential conflicts due to existing use of field name
```

Comment:
That's managed by registering the header field, and is well-documented in HTTP. Why is this important to call out here?

Reply:
The example I had in mind were the conflicts @mikewest ran into when deploying the initial version of `upgrade-insecure-requests` as `https`. I believe that resulted in some server side conflicts with some servers that already used that name internally, and read all incoming request headers to be global variables. (@mikewest - please correct me if I remember that wrong)

Reply:
Some versions of PHP do strange things with certain headers. `https` was one of them. I'm not sure that has much weight in either direction, except as an anecdote that gave me a very bad week a few years ago. :)

Reply:
Sure. I just don't know why we're doing this specifically here, rather than relying on the generic HTTP text (DRY and all that). Not a huge deal, just struck me as odd.

Reply:
So Yoav explained that the main reason here is that the reason for the Sec- prefix was to avoid problems like the one where header fields are turned into environment variables in a non-injective fashion by some frameworks.  So you get the value of the "Foo" header field mixed in with the "foo" query parameter.  That motivates **a** prefix, but it doesn't motivate the "Sec-" one.

I hadn't heard that reason properly before.  It took me a while to internalize it.

And then it didn't make sense.  The more I think about this, the more that I think that the "CH-" prefix is useful, but the "Sec-" one isn't.  On the contrary, I see value in giving script the ability to query for different content using these header fields.  For one, it might allow an app that is preparing for going offline in the browser the ability to request content based on anticipated rather than current needs.

For example, assuming for a moment that I thought `Save-Data` was useful (it's not), then the content could request the lightweight content that this targets, to avoid expending resources on the fetch or storage.  At that point, I don't have to have an opinion on `Save-Data` from the browser perspective, it might as well be yet another proprietary header field.